File: heap/leetcode_2974/solution.py
from typing import List

class Solution:
    def numberGame(self, nums: List[int]) -> List[int]:
        if not nums:
            return None

        result = []

        nums.sort()

        while nums:
            first_num = nums.pop(0)
            second_num = nums.pop(0) if nums else None

            result.append(second_num)
            result.append(first_num)

        return result

File: heap/leetcode_2974/solution.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var numberGame = function(nums) {
    if(!nums.length) {
        return null;
    }

    let result = [];

    nums.sort((a, b) => a - b);

    while(nums.length) {
        const firstNum = nums.shift();
        const secondNum = nums.shift();

        result.push(secondNum);
        result.push(firstNum);
    }

    return result;
};